source/NVDAObjects/window/excel.py

```diff
@@ -879,29 +905,49 @@ def _get_states(self):
 		return states
 
 	def script_changeSelection(self,gesture):
-		oldSelection=api.getFocusObject()
-		gesture.send()
-		import eventHandler
-		import time
-		newSelection=None
-		curTime=startTime=time.time()
-		while (curTime-startTime)<=0.15:
-			if scriptHandler.isScriptWaiting():
-				# Prevent lag if keys are pressed rapidly
-				return
-			if eventHandler.isPendingEvents('gainFocus'):
-				return
-			newSelection=self._getSelection()
-			if newSelection and newSelection!=oldSelection:
-				break
-			api.processPendingEvents(processEventQueue=False)
-			time.sleep(0.015)
-			curTime=time.time()
-		if newSelection:
-			if oldSelection.parent==newSelection.parent:
+		# Detect if this script is being repeated a lot (E.g. held down)
+		repeated=scriptHandler.getLastScriptRepeatCount()>2
+		if repeated and scriptHandler.isScriptWaiting():

```

Comment:
Previously this was done in the while loop until the timeout, is this moved out since we expect this script to get called again? How do we know that this script will be called again?